cpp/Contact.cpp

```diff
@@ -114,7 +116,8 @@ std::size_t dolfinx_contact::Contact::coefficients_size()
   const std::size_t bs = dofmap->bs();
 
   // NOTE: Assuming same number of quadrature points on each cell
-  const std::size_t num_q_points = _qp_ref_facet[0].shape(0);
+  const std::size_t num_q_points
+      = _quadrature_rule->offset()[1] - _quadrature_rule->offset()[0];
```

Comment:
Maybe add an if test on the cell_type here, i.e. not prism or pyramid. As we at some point might revisit it and it would be easier to catch functions where we have made this assumption.